lib/fs.js

```diff
@@ -1562,6 +1562,8 @@ fs.unwatchFile = function(filename, listener) {
   }
 };
 
+// Cache for JS real path 
+var realpathCache = {};
```

Comment:
One global `realpath` cache without some kind of cache validation is probably a no-go?